### PR TITLE
Add definition of nRF52805 SoC and nrf52dk_nrf52805 board

### PR DESCRIPTION
**Pull request by anangl**
Add definition of the nRF52805 SoC together with the corresponding dts files.

### CODE REVIEW

**Review by ioannisg — COMMENTED** [review]
Looks ok to me, didn't spot any issues, however, I would like to see a board definition, if possible

File: boards/arm/nrf52dk_nrf52805/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# The nrf52dk_nrf52805 board mirrors the nRF52832 DK hardware. This
```
Comment (ioannisg) [inline boards/arm/nrf52dk_nrf52805/CMakeLists.txt:3]: Strictly speaking, there is no "nRF52832 DK" hardware. I know we have the same line in nrf52810, but being pedantic, this should be nRF 52 DK

**Review by ioannisg — APPROVED** [review]
Looks fine to me

File: dts/arm/nordic/nrf52805.dtsi [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+			#size-cells = <0>;
+			reg = <0x40003000 0x1000>;
+			clock-frequency = <I2C_BITRATE_STANDARD>;
+			interrupts = <3 1>;
+			status = "disabled";
+			label = "I2C_0";
+		};
+
+/* Disabled until MDK is updated to provide QDEC definition and bitfields. */
+#if 0
```
Comment (ioannisg) [inline dts/arm/nordic/nrf52805.dtsi:104]: Why was this needed? Did it result in an build errors?

**Review by ioannisg — COMMENTED** [review]
@anangl the nrf52805 should be "CAAA" not QFAA

File: dts/arm/nordic/nrf52805.dtsi (hunk 2 above)
Comment (anangl) [reply dts/arm/nordic/nrf52805.dtsi:104]: Yes, the build assertion checking the base address failed here:
https://github.com/zephyrproject-rtos/zephyr/blob/b046ca54093abbede73d507afa661a8470dd352a/soc/arm/nordic_nrf/validate_base_addresses.c#L104

File: boards/arm/nrf52dk_nrf52805/CMakeLists.txt (hunk 1 above)
Comment (anangl) [reply boards/arm/nrf52dk_nrf52805/CMakeLists.txt]: Fixed.

**Comment by anangl** [comment]
> @anangl the nrf52805 should be "CAAA" not QFAA

Fixed.

**Review by ioannisg — APPROVED** [review]

**Review by ioannisg — CHANGES_REQUESTED** [review]
One more question @anangl , you do not seem to introduce an LE Radio header for this SoC. I wonder how this actually works, do the LE samples build?

**Comment by cvinayak** [comment]
> 
> 
> One more question @anangl , you do not seem to introduce an LE Radio header for this SoC. I wonder how this actually works, do the LE samples build?

If the BLE Radio headers have same values as that of nRF52832, then only a conditional include of existing nRF52832 header should suffice. nRF52 Series is starting the get variants with same Radio peripheral/characteristics, duplicate header files can be avoided.

**Comment by carlescufi** [comment]
Please update the SHA in west.yml

**Comment by anangl** [comment]
> Please update the SHA in west.yml

Done.

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
> #### Conflicting files
> `west.yml`

needs rebase

**Comment by anangl** [comment]
Rebased on master.

**Comment by anangl** [comment]
DNM until BLE support is added.

**Comment by anangl** [comment]
@cvinayak Could you take a look at the Bluetooth related stuff (last two commits)?

**Review by cvinayak — APPROVED** [review]
Looks good to me. Have tested the samples/bluetooth/peripheral built for nrf52dk_nrf52805 on nrf52dk_nrf52832 with increased RAM manually in Kconfig. No issues noticed with the changes to use PPI 2, 3 and 4 for sw tIFS switch (instead of normally used 10, 11 and 12 on nRF52832).

**Comment by carlescufi** [comment]
@ioannisg please re-review

File: soc/arm/nordic_nrf/nrf52/Kconfig.defconfig.nrf52805_QFAA [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Nordic Semiconductor nRF52805 MCU
```
Comment (ioannisg) [inline soc/arm/nordic_nrf/nrf52/Kconfig.defconfig.nrf52805_QFAA:1]: @anangl shouldn't this file be named with the right suffix, CAAA?

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h [hunk 4]
```diff
@@ -369,7 +369,15 @@ static inline void hal_sw_switch_timer_clear_ppi_config(void)
 /* Wire the RADIO EVENTS_END event to one of the PPI GROUP TASK ENABLE task.
  * 2 adjacent PPI groups are used for this wiring. 'index' must be 0 or 1.
  */
+#if defined(CONFIG_SOC_NRF52805)
+#if (EVENT_TIMER_ID == 0)
+#define HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI 2
+#else
+#error "Insufficient resources"
```
Comment (ioannisg) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h:376]: I guess you should explain the reason for the "insufficient resources", either in a comment, or directly in the preprocessor #error 

Hard to see how the insufficient resources relates to the ID of the Event TImer.

**Comment by ioannisg** [comment]
> @ioannisg please re-review

Just did that @carlescufi 
I've just got 2 comments/questions for @anangl then, I guess, it's good to go.

File: soc/arm/nordic_nrf/nrf52/Kconfig.defconfig.nrf52805_QFAA (hunk 3 above)
Comment (anangl) [reply soc/arm/nordic_nrf/nrf52/Kconfig.defconfig.nrf52805_QFAA:1]: Of course it should. Thanks for catching this!

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h (hunk 4 above)
Comment (anangl) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h:376]: I'll add some more descriptive message/comment.

Comment (anangl) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h]: Added.

File: soc/arm/nordic_nrf/nrf52/Kconfig.defconfig.nrf52805_QFAA (hunk 3 above)
Comment (anangl) [reply soc/arm/nordic_nrf/nrf52/Kconfig.defconfig.nrf52805_QFAA]: Fixed.

**Review by ioannisg — APPROVED** [review]
thanks!
